lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -130,20 +130,19 @@ def ortho_transformation(zfront, zback):
 
 def proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here..
-    txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
-    return txs, tys, tzs
+    vecw /= vecw[3]
+    return vecw[0:3]
 
 def proj_transform_vec_clip(vec, M):
     vecw = np.dot(M, vec)
-    w = vecw[3]
-    # clip here.
-    txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
+    # Determine clipping before rescaling
     tis = (0 <= vecw[0]) & (vecw[0] <= 1) & (0 <= vecw[1]) & (vecw[1] <= 1)
     if np.any(tis):
         tis = vecw[1] < 1
-    return txs, tys, tzs, tis
+    vecw /= vecw[3]
+    # Integrating tis in the numpy array for optimization purposes
+    vecw[3, :] = tis
```

Comment:
https://github.com/matplotlib/matplotlib/pull/6085#discussion_r128393125 by @eric-wieser 

> Get rid of this colon - it makes the function fail on 1d inputs

I have not changed it  as I think this will work with scalar and 1D arrays, also CI pass. Unless I'm missing something?

Reply:
If `vecw.ndim == 1`, then `vec[3,:]` is an error

Reply:
I understand - fixed.